### ISSUE TITLE
Migrate import above size with exclusions does not update .gitAttributes properly

### ISSUE THREAD

**Issue by jrcmramos** [git-lfs/git-lfs#4769]
**Description**
While using the migrate import tool with the `--above` and `--exclude` options,  the `.gitAttributes` file is not updated properly.
All migrated files are not added to the list.

**To Reproduce**
Steps to reproduce the behaviour:
1. Setup a repository with more than 1 different file type more than 1MB (pe `jpg` and `png`)
2. Run the migrations
```
git lfs migrate import --above=1Mb --exclude="*.jpg" --everything --yes
```
3. Only the excluded file is added to `.gitAttributes`. In this case
```
*.jpg !text -filter -merge -diff
```

**Expected behavior**
All files migrated listed in the `.gitAttributes` file.

**System environment**
MacOS 11.3.1 
Git 2.31.1
Git LFS 3.0.1

**Comment by bk2204** [comment]
Hey,

Thanks for the report.  The `--above` option is currently not functional with `--include` and `--exclude`, and we should accurately reflect that in the code and the documentation.  We'd be happy to accept a patch to make them work together, but at the moment, it's a bug that they currently do.

**Comment by bk2204** [comment]
I've opened #4771 to fix this problem where we accept this combination.  Therefore, I'm going to retitle this to reflect that these options should work together and then mark it as an enhancement.

**Comment by jrcmramos** [comment]
Thank you for the fast feedback @bk2204 ! 🙇‍♂️

**Comment by Quuxplusone** [comment]
Just to clarify: Are you saying that this line has never worked and doesn't work now either?
```
git lfs migrate import --everything --include='*.zip' --above=100KB
```
I'd expect this line to import into LFS every `.zip` file larger than 100KB. My copy of `git-lfs` (i.e. `git-lfs/3.0.2 (GitHub; linux amd64; go 1.18.1)`) accepts this line and does _something_ with it. But you're saying maybe it doesn't do the right thing?

**Comment by chrisd8088** [comment]
> But you're saying maybe it doesn't do the right thing?

That's correct, it didn't always do what a user would intuitively expect.  The original report notes that the `--exclude` and `--above` options did not produce the desired result, and we can illustrate one of the problems when using the `--include` and `--above` options together with a short example:

```
$ git init test
$ cd test
$ echo foo >foo.bin
$ echo foobar >foobar.bin
$ git add *.bin
$ git commit -m init

# Using a Git LFS version prior to 3.1.0:
$ git lfs migrate import --include="*.bin" --above=5B

$ cat .gitattributes 
*.bin filter=lfs diff=lfs merge=lfs -text

$ git lfs ls-files
aec070645f - foobar.bin

$ git lfs fsck
pointer: unexpectedGitObject: "foo.bin" (treeish fc18869f009bc55cd1f6d62a3f34316a78b7e6c4) should have been a pointer but was not
```

So the smaller file `foo.bar` is skipped by the `git lfs migrate import` command because it falls below the size limit specified with the `--above` option, but the `--include` option caused the `.gitattributes` file to be created with an entry that matches all `*.bin` files, including the smaller `foo.bar` file.  That will just cause problems later because the `.gitattributes` entry means all `*.bin` files should be Git LFS objects, but only some of them have been converted from regular files.

(Note that the `--everything` option is unrelated to this issue.  It just indicates to the `git lfs migrate import` command that the command should operate on all branches and not just the current branch.)

If we run the `git lfs migrate import` command with the `--above` option but without the `--include` option, we get the intended result, where the `.gitattributes` file only matches the files that have been converted to Git LFS objects:

```
$ git init test
$ cd test
$ echo foo >foo.bin
$ echo foobar >foobar.bin
$ git add *.bin
$ git commit -m init

$ git lfs migrate import --above=5B

$ cat .gitattributes 
/foobar.bin filter=lfs diff=lfs merge=lfs -text

$ git lfs ls-files
aec070645f - foobar.bin

$ git lfs fsck
Git LFS fsck OK
```
